### ISSUE TITLE
Performance Issue with List<> & Prepend

### ISSUE THREAD

**Issue by RealDotNetDave** [dotnet/coreclr#25701]
I benchmarked creating a collection uisng Prepend for a generic list between Core 3 and Clr 4.8. In all of the tests, Core 3 is slower.

![image](https://user-images.githubusercontent.com/15676130/61247456-fc3c5c00-a705-11e9-959b-5f86b70019f4.png)

Below is the benchmark code:
```
        public List<FakePerson> TestCollectionCreating()
        {
            var collection = new List<FakePerson>();

            for (int count = 0; count < this.CollectionCount; count++)
            {
                collection.Prepend(this._fakePersonFixedCollection[count]);
            }

            return collection;
        }
```

**Comment by stephentoub** [comment]
@RealDotNetDave, what is `Prepend`?  No such method exists on `List<T>`.  Is this LINQ's Prepend?  If so, Prepend doesn't modify its input, and rather returns a new enumerable that contains the additional item, so the benchmark isn't actually testing what it's meant to be testing.

Please include a complete, standalone repro, i.e. something I can literally copy and paste into a new .cs file and run.  This example is using types that aren't included, is missing things `[Benchmark]`, is missing the various inputs used to execute it, etc.

**Comment by RealDotNetDave** [comment]
You are right, not sure if I would use this to create a collection. Let me change the test to add one item then return the collection. Will update this. Thanks!

**Comment by safern** [comment]
> Will update this. Thanks!

@RealDotNetDave any update?
